### ISSUE TITLE
Consider adding Result.requireNonNull

### ISSUE THREAD

**Issue by danyx23** [demystifyfp/FsToolkit.ErrorHandling#55]
I really like this library! I just found myself wanting a function that takes a nullable value and makes sure that it is not null by turning it into an Error value if it is null (useful for the many dotnet function that might return null). Would you consider a PR that adds something like the code below?

```fsharp
let requireNotNull (errorValue : 'error) (value : 'a when 'a : null ) =
    match value with
    | null -> Error errorValue
    | nonnull -> Ok nonnull
```

**Comment by cmeeren** [comment]
Would `Option.ofObj >> Result.requireSome` suit your purposes?

**Comment by demystifyfp** [comment]
@cmeeren It'd help. But I feel @danyx23 requirement makes sense in the context of other such helper functions. What do you recommend? 

One suggestion on the name of the function, can we have it as `requireNonNull`?

**Comment by cmeeren** [comment]
The helper function makes sense to me. 🙂

I'd call it requireNotNull (as OP suggested) instead of requireNonNull though, to fit with other functions (e.g. notEqualTo) (and in my mind it sounds better), but I certainly don't want to create a fuss over the difference.

**Comment by danyx23** [comment]
Yes I think while `Option.ofObj >> Result.requireSome` is also a nice way it would be good to have something specific for this rather common case that is a bit more obvious when reading the code. 

I am totally fine with requireNotNull. I'll send a PR and then we can discuss any further details there. Thansk!

**Comment by danyx23** [comment]
Sent a PR. Let me know what you think. I only realized in the PR that there are some whitespace changes because my editor is set to remove trailing whitespace and doesn't show whitespace diffs when staging changes - hope you don't mind.

**Comment by demystifyfp** [comment]
Thanks, @danyx23 for the PR. I've merged the changes and published the new version `1.2.3`. Can you verify the changes and close this issue?

**Comment by danyx23** [comment]
Looks good. I have some strange issue with the update of the nuget file in the project that I am using this but the assembly has the new code and everything should work so I'm closing the issue.
